nemo/collections/tts/models/fastpitch.py

```diff
@@ -404,13 +438,18 @@ def training_step(self, batch, batch_idx):
         return loss
 
     def validation_step(self, batch, batch_idx):
-        attn_prior, durs, speaker = None, None, None
+        attn_prior, durs, speaker, energy = None, None, None, None
         if self.learn_alignment:
             if self.ds_class_name == "TTSDataset":
-                if SpeakerID in self._train_dl.dataset.sup_data_types_set:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _, speaker = batch
-                else:
-                    audio, audio_lens, text, text_lens, attn_prior, pitch, _ = batch
+                batch_dict = process_batch(batch, self._train_dl.dataset.sup_data_types_set)
+                audio = batch_dict.get("audio")
+                audio_lens = batch_dict.get("audio_lens")
+                text = batch_dict.get("text")
+                text_lens = batch_dict.get("text_lens")
+                attn_prior = batch_dict.get("align_prior_matrix", None)
+                pitch = batch_dict.get("pitch", None)
+                energy = batch_dict.get("energy", None)
+                speaker = batch_dict.get("speaker_id", None)
             else:
                 raise ValueError(f"Unknown vocab class: {self.vocab.__class__.__name__}")
         else:
```

Comment:
Same question as the above. I wonder if we can make the logic more concise like below, @rlangman could you pls confirm?

``` python
        ...
        batch_dict = process_batch(batch, self._train_dl.dataset.sup_data_types_set)
        audio = batch_dict.get("audio")
        audio_lens = batch_dict.get("audio_lens")
        text = batch_dict.get("text")
        text_lens = batch_dict.get("text_lens")
        attn_prior = batch_dict.get("align_prior_matrix", None)
        pitch = batch_dict.get("pitch", None)
        energy = batch_dict.get("energy", None)
        speaker = batch_dict.get("speaker_id", None)
        dur = batch_dict.get("dur", None)

        if self.learn_alignment:
            assert self.ds_class_name == "TTSDataset", f"Unknown vocab class: {self.vocab.__class__.__name__}"
```

Reply:
Sure, we can but the assert statement has to go before we start extracting elements from the dict. Because if `ds_class_name != "TTSDataset"` then the extraction code would throw error and we will not exit gracefully.